Update documentation to reflect additions in RMG-database

kspieks opened the pull request:
This minor PR updates the documentation that tabulates which levels of theory are supported by Arkane. It reflects changes from a recent RMG-database [PR](https://github.com/ReactionMechanismGenerator/RMG-database/pull/508), which expanded the supported AEC atom types for coupled cluster calculations with `CCSD(T)-F12/cc-pVDZ-F12` and `CCSD(T)-F12/cc-pVTZ-F12`.

kspieks commented:
This PR also fixes a bug with `arkane/encorr/bac.py` that prevented it from properly overwritting existing BAC values in RMG-database.

 `arkane/encorr/bacTest.py` is supposed to test that the previous `"LevelOfTheory(method='ccsd(t)f12',basis='ccpvtzf12',software='molpro')"` [entry](https://github.com/ReactionMechanismGenerator/RMG-database/blob/master/input/quantum_corrections/data.py#L472-L479) can be overwritten by the dummy test [values](https://github.com/ReactionMechanismGenerator/RMG-Py/blob/master/arkane/encorr/bacTest.py#L73). However, since `arkane/encorr/bac.py` has a logic [error](https://github.com/ReactionMechanismGenerator/RMG-Py/blob/master/arkane/encorr/bac.py#L825) while parsing `input/quantum_corrections/data.py`, it will accidentally overwrite the first instance of `"LevelOfTheory(method='ccsd(t)f12',basis='ccpvtzf12',software='molpro')"` , which happens to be the new `"CompositeLevelOfTheory(freq=LevelOfTheory(method='wb97xd3',basis='def2tzvp',software='qchem'),energy=LevelOfTheory(method='ccsd(t)f12',basis='ccpvdzf12',software='molpro'))"` entry for the new RMG-database [PR](https://github.com/ReactionMechanismGenerator/RMG-database/pull/508) instead. This ultimately causes the unit test to fail since it is expecting `"LevelOfTheory(method='ccsd(t)f12',basis='ccpvtzf12',software='molpro')"` to have been overwritten. Since that PR is the first to add composite LoT to RMG-database, it has found this bug :)

This PR fixes this error and updates `bacTest.py` to additionally test that `CompositeLevelOfTheory` entries are ignored when relevant and overwitten when desired.

@xiaoruiDong Note that the additional test causes this PR's tests to [fail](https://github.com/ReactionMechanismGenerator/RMG-Py/pull/2175/checks?check_run_id=3042694169#step:9:5961) due to 
```
arkane.exceptions.BondAdditivityCorrectionError: No BACs available for writing
```

which is expected since no composite LoT currently exists in RMG-database to overwrite. This unittest will not pass until the twin RMG-database [PR](https://github.com/ReactionMechanismGenerator/RMG-database/pull/508) is merged. However, the tests for that PR will not pass until this PR is merged. I've checked out the `update_ccsdtf12` branch of RMG-database and ` update_model_chem_doc` from RMG-Py and run `make test test-unittests` locally. These tests pass, so these twin PRs are compatible with each other. But one will need to be merged in first

xiaoruiDong commented:
Thanks, @kspieks. I just finished my review. My laptop was dead last week, so the review was postponed... Sorry about that. I am fine with your implementations. I added another commit... Just find there is a 5-layer-if-else + a 2 layer for combined together near your modifications, which is not friendly for reviewing. So I refactored the codes and added a few comments. I restarted both the test for RMG-Py and RMG-test for your PRs, and I will let you know once it finishes. If all checks pass, please remove the temporary commits and rebase the branch. I will help you merge them afterward.

xiaoruiDong commented:
@kspieks The only failure test is by Travis, which is okay. Feel free to check my modifications and let me know if you have more to add. Otherwise, please rebase the branches, remove the temporary Git Action commit, and @me to merge your twin PRs. Thanks!

kspieks commented:
@xiaoruiDong Thanks for your review and hope your laptop is fine now. I appreciate you making that code section more clear :) It looks good to me so I rebased and removed the temporary git action commit on both this PR and the PR for RMG-database
